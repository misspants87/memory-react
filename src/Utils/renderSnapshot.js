import React from 'react';
import renderer from 'react-test-renderer';

const renderSnapshot = (component) => {
  return renderer.create(component).toJSON();
}

export default renderSnapshot;
