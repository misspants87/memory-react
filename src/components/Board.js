import React, { useEffect} from 'react';
import Square from './Square.js';
// const uuid = require('uuid/v4');

const Board = props => {


//if tempExposedSquares grows in length, then we know exposedSquares won't change
//if exposedSquares grows in length, then we can keep showing tempExposedSquares

//map through squares, you need a square for each square
//if the index of the square matches an index found in exposedSquares or tempExposedSquares, then return an image
//otherwise, return an empty box

console.log('board');

let exposedSquares = new Map();


  props.exposedSquares.map((square) => {
    exposedSquares.set(square, true);
  });

let grid =  () => {

  console.log(exposedSquares)
  return props.squares.map((square, index) => {
    if (exposedSquares.has(index)) {
      return (
        <Square
          onClick = {props.onClick.bind(this, index, square)}
          square = {square}
          index = {index}
          key ={index}
          showImage = {true}
        />
      )
    } else {
      return (
        <Square
          onClick = {props.onClick.bind(this, index, square)}
          key ={index}
          square = {square}
          index = {index}
          showImage = {false}
        />
      )
    }
  })
}

  return (
    <div className="container">
      {grid()}
    </div>
  )
}

export default React.memo(Board, (prevProps, nextProps) => {
  return (
    prevProps.exposedSquares === nextProps.exposedSquares &&
    prevProps.squares === nextProps.squares
  )
});
