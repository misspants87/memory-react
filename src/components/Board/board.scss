.grid-container {
  height: 275px;
  width: 600px;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr 1fr 1fr;
}
