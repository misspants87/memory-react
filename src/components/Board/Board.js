import React, { useEffect, useCallback} from 'react';
import Square from '../Square/Square.js';
import PropTypes from 'prop-types';
import './board.scss';

const Board = props => {

//map through squares, you need a square for each square
//if the index of the square matches an index found in exposedSquares or tempExposedSquares, then return an image
//otherwise, return an empty box

//useCallback memoizes callbacks, to prevent unnecessasary renderings
// const onClick = useCallback(
//   (index, square) => {

//     return props.onClick(index, square)
//   },
//   [props.onClick]
// )


let squaresToShow = new Map();

  props.exposedSquares.map((square) => {
    squaresToShow.set(square, true);
  });

let grid = () => {

  return props.squares.map((square, index) => {
      return (
        <Square
          onClick = {props.onClick.bind(this, index, square)}
          square = {square}
          index = {index}
          key ={index}
          showImage = {squaresToShow.has(index)}
          data-test="Board Component"
        />
      )
  })
}

  return (
    <div className="grid-container">
      {grid()}
    </div>
  )
}

//this is causing error in browser console if wrong prop type is passed in to board
//problem is in tests, when using check-prop-types library, no error or warning thrown
Board.propTypes = {
  squares: PropTypes.array,
  exposedSquares: PropTypes.array,
  whichPlayer: PropTypes.string,
  onClick: PropTypes.func,
}

//React.memo prevents checkPropTypes from correctly validating prop types. Likely since component has been memoized with correct prop types
export default Board;
// export default React.memo(Board, (prevProps, nextProps) => {
//   return (
//     prevProps.exposedSquares === nextProps.exposedSquares &&
//     prevProps.squares === nextProps.squares
//   )
// });
