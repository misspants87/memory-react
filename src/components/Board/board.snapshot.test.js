import React from 'react';
import Board from './Board';
import renderSnapshot from '../../Utils/renderSnapshot';

test('should render Board Component correctly', () => {
  let snapshot = renderSnapshot(<Board
    squares={[47,47]}
    exposedSquares = {[1]}
    whichPlayer = {'A'}
    onClick = {() => 'Fake fn'}
  />)

  expect(snapshot).toMatchSnapshot();
})
