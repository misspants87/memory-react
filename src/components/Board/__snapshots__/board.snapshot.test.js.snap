// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Board Component correctly 1`] = `
<div
  className="container"
>
  <button
    data-test="Square Component"
    onClick={[Function]}
  />
  <button
    data-test="Square Component"
    onClick={[Function]}
  >
    <img
      alt=""
      src={47}
    />
  </button>
</div>
`;
